frontend/packages/core/src/flags.tsx

```diff
@@ -0,0 +1,115 @@
+import * as React from "react";
+import type { clutch as IClutch } from "@clutch-sh/api";
+
+import type { ClutchError } from "./Network/errors";
+import { client } from "./Network/index";
+
+const FF_POLL = +process.env.REACT_APP_FF_POLL || 300000;
+const FF_CACHE_TTL = +process.env.REACT_APP_FF_CACHE_TTL || 60000;
+
+export interface FeatureFlags {
+  [name: string]: IClutch.featureflag.v1.IFlag;
+}
+
+const featureFlags = (): Promise<FeatureFlags> => {
+  const cachedFlags = JSON.parse(sessionStorage.getItem("featureFlags"));
+  if (cachedFlags) {
+    const elapsedTime = new Date(new Date().getTime() - cachedFlags.timestamp).getTime();
+    if (elapsedTime < FF_CACHE_TTL) {
+      return new Promise(resolve => resolve(cachedFlags.flags));
+    }
+  }
+  return client
+    .post("/v1/featureflag/getFlags", {}, { timeout: 1000 })
```

Comment:
does `client` set a default timeout?

Reply:
yeah, we rely on the gateway timeout instead of on the client. This is set here intentionally since we don't want feature flags to become a blocking call